File: Clases_web/Vanetas/src/au/cl/gm/ventas/Producto.java
package au.cl.gm.ventas;

public class Producto {
    private final int idProducto;
    private String nombre;
    private double precio;
    private static int contadorProductos;
    
    private Producto(){
        //por cada objeto que se cree tenemos un nuevo id
        //que se asigna a idProducto
        this.idProducto = ++Producto.contadorProductos;
        
    }

    public Producto(String nombre, double precio) {
        //llamada al constructor privado
        //asi se inicializa y da un id
        this();
        this.nombre = nombre;
        this.precio = precio;
    }

    public int getIdProducto() {
        return idProducto;
    }
    
    public String getNombre() {
        return this.nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public double getPrecio() {
        return this.precio;
    }

    public void setPrecio(double precio) {
        this.precio = precio;
    }

    @Override
    public String toString() {
        StringBuilder sb = new StringBuilder();
        sb.append("Producto{");
        sb.append("idProducto=").append(idProducto);
        sb.append(", nombre=").append(nombre);
        sb.append(", precio=").append(precio);
        sb.append('}');
        return sb.toString();
    }
    
    
    
}

File: JavaScript/SistemaVentas/Producto.js
class Producto{
    static contadorProductos = 0;

    constructor(nombre, precio){
        this._idProducto = ++Producto.contadorProductos;
        this._nombre = nombre;
        this._precio = precio;
    }

    get idProducto(){
        return this._idProducto;
    }
    get nombre(){
        return this._nombre;
    }
    set nombre(nombre){
        this._nombre = nombre
    }
    get precio(){
        return this._precio
    }
    set precio(precio){
        this._precio = precio;
    }

    toString(){
        return `
            idProducto: ${this._idProducto}\n
            nombre: ${this._nombre}\n
            precio: $${this._precio}
        `
    }
}

let producto1 = new Producto('Jugo', 1000);
let producto2 = new Producto('Naranja', 20);

console.log(producto1.toString());
console.log(producto2.toString())